src/display/canvas.js

```diff
@@ -2062,7 +2062,20 @@ class CanvasGraphics {
     }
 
     let patternFillTransform, patternStrokeTransform;
-    if (current.patternFill) {
+
+    // Only compute pattern transforms if the text rendering mode actually
+    // uses fill/stroke. This avoids expensive pattern calculations each call
+    // when a patternFill/patternStroke is set, but unused.
+    const fillStrokeMode =
+      current.textRenderingMode & TextRenderingMode.FILL_STROKE_MASK;
+    const needsFill =
+      fillStrokeMode === TextRenderingMode.FILL ||
+      fillStrokeMode === TextRenderingMode.FILL_STROKE;
+    const needsStroke =
+      fillStrokeMode === TextRenderingMode.STROKE ||
+      fillStrokeMode === TextRenderingMode.FILL_STROKE;
+
+    if (current.patternFill && needsFill) {
```

Comment:
nit: put `needsFill` in the first position: it's slightly faster to test the boolean first.